SU2_CFD/src/solvers/CSolver.cpp

```diff
@@ -2456,8 +2456,8 @@ void CSolver::SetRotatingFrame_GCL(CGeometry *geometry, const CConfig *config) {
 
 void CSolver::SetAuxVar_Gradient_GG(CGeometry *geometry, const CConfig *config) {
 
-  const auto solution = base_nodes->GetAuxVar();
-  auto gradient = base_nodes->GetAuxVarGradient();
+  const auto& solution = base_nodes->GetAuxVar();
+  auto& gradient = base_nodes->GetAuxVarGradient();
```

Comment:
indeed the not so auto thing about auto...

Reply:
Is there no regression for axisymmetric flow? One would think something like this could break something.

Reply:
I don't think there is but it did just set them to zero and we don't divide by them or anything so nothing broke just inaccurate results and perhaps not converging when it should in some cases. No compiler error or warning so i guess with auto one needs to be careful

Reply:
Can we have a regression that "exercises" this part of the implementation? Perhaps just modifying an existing case?

Reply:
@FlorianDm @pcarruscag  believe there was another issue within NEMO axisymmetric that @jtneedels fixed.  Maybe it would be a good time to merge that in as well?

Reply:
@FlorianDm Yes I agree with these changes, thanks for catching that! @pcarruscag Yes, as @WallyMaier mentioned I do have another small fix to pull in for axisymmetric in CNEMOEulerSolver, should I go ahead ad open a separate PR to merge that in?

Reply:
I will try to create a test/regression case after adding the k-omega source terms (almost done) because I think it is more useful if it is a rans case